### PR TITLE
Add webnn reshape tests

### PR DESCRIPTION
**Pull request by BruceDai**
@fdwr @Honry @huningxin @dontcallmedom PTAL, thanks.

### CODE REVIEW

File: webnn/resources/test_data/reshape.json [hunk 1]
```diff
@@ -0,0 +1,272 @@
+{
+  "tests": [
```
Comment (fdwr) [inline webnn/resources/test_data/reshape.json:2]: How about one test case coming from or going to 1D? e.g. `"newShape": [24],`

Comment (fdwr) [reply webnn/resources/test_data/reshape.json:2]: And/or `"newShape: [-1]` which would consolidate all elements into a single 1D tensor. I don't really care which, but I think we should have at least one 1D test case.

Comment (BruceDai) [reply webnn/resources/test_data/reshape.json:2]: Thanks, I've added such tests. Please take another look.

**Review by fdwr — APPROVED** [review]

**Comment by BruceDai** [comment]
@Honry All checks passed, PTAL, thanks.

**Review by Honry — APPROVED** [review]
LGTM, thanks!

**Pull request merged**